examples/Examples/ByteString.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE TypeApplications    #-}
+
+module Examples.ByteString (
+    exampleByteStringAnd,
+    exampleByteStringOr
+  ) where
+
+import           Data.Data                                   (Proxy (Proxy))
+import           Data.Function                               (($))
+import           Data.List                                   ((++))
+import           Data.String                                 (String)
+import           GHC.TypeNats                                (KnownNat, natVal)
+import           System.IO                                   (IO, putStrLn)
+import           Text.Show                                   (show)
+
+import           ZkFold.Base.Algebra.Basic.Class
+import           ZkFold.Base.Algebra.Basic.Field             (Zp)
+import           ZkFold.Base.Algebra.EllipticCurve.BLS12_381 (BLS12_381_Scalar)
+import           ZkFold.Symbolic.Compiler                    (ArithmeticCircuit, compileIO)
+import           ZkFold.Symbolic.Data.ByteString
+
+exampleByteStringAnd :: forall n . KnownNat n => IO ()
+exampleByteStringAnd = makeExample @n "*" "and" (*)
+
+exampleByteStringOr :: forall n . KnownNat n => IO ()
+exampleByteStringOr = makeExample @n "+" "or" (+)
+
+type Binary a = a -> a -> a
```

Comment:
Should we move this definition to the lib?

Reply:
I guess not as it is only used for brevity and only in examples
Maybe later, when we see the need for a general approach to preparing examples